### PR TITLE
Xml comments support in synapse configuration

### PR DESCRIPTION
**Pull request by jagathsisira**
With this updated, selected XML configurations will preserve XML comments in ESB configuration files.

### CODE REVIEW

File: components/mediation-admin/org.wso2.carbon.mediator.service/src/main/java/org/wso2/carbon/mediator/service/builtin/CommentMediator.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        return parent;
+    }
+
+    /**
+     * Build the mediator with given OMElement node
+     *
+     * @param elem synapse mediator configuration
+     */
+    public void build(OMElement elem) {
+    }
```
Comment (prabathabey) [inline components/mediation-admin/org.wso2.carbon.mediator.service/src/main/java/org/wso2/carbon/mediator/service/builtin/CommentMediator.java]: Maybe it's a good idea to throw an UnsupportedMethodException or something similar here in this particular method?

File: components/mediation-ui/org.wso2.carbon.sequences.ui/src/main/java/org/wso2/carbon/sequences/ui/util/SequenceEditorHelper.java [hunk 2]
```diff
@@ -460,4 +483,44 @@ private static String getEditorMode(EditorUIClientFactory factory) {
...
     }
+
+    /**
+     * Check the validity of a given mediator to be shown in UI
+     *
+     * @param mediator AbstractListMediator instance to be checked
+     * @return true if valid, false otherwise
+     */
+    private static boolean isValidMediatorForUI(Mediator mediator) {
+        boolean validity = true;
```
Comment (prabathabey) [inline components/mediation-ui/org.wso2.carbon.sequences.ui/src/main/java/org/wso2/carbon/sequences/ui/util/SequenceEditorHelper.java]: Can further simplify contents in this particular method body as, "return !(mediator instanceof CommentMediator)"

File: components/mediation-config-admin/org.wso2.carbon.mediation.configadmin/src/main/java/org/wso2/carbon/mediation/configadmin/util/XMLPrettyPrinter.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    private boolean xmlFormat;
+    private boolean numericEnc;
+    private boolean done = false;
+    private String encoding = "UTF-8";
+
+    private static Log log = LogFactory.getLog(XMLPrettyPrinter.class);
+
+    public XMLPrettyPrinter(InputStream in, boolean format, boolean numeric, String encoding) {
+        this.in = in;
+        xmlFormat = format;
```
Comment (lafernando) [inline components/mediation-config-admin/org.wso2.carbon.mediation.configadmin/src/main/java/org/wso2/carbon/mediation/configadmin/util/XMLPrettyPrinter.java]: Be consistent on how you assign instance variable values here, if you used "this.in = in", then it should be "this.format = format" ..

File: components/mediation-ui/org.wso2.carbon.sequences.ui/src/main/resources/web/sequences/design_sequence.jsp [hunk 4]
```diff
@@ -463,13 +463,13 @@
...
-    function afterMediatorUpdate(src) {
-        var error  = src.trim();
-        if(error != '' && (error.indexOf("error:") != -1)){
-            CARBON.showErrorDialog(error.substring(6,error.length));
-            if (document.getElementById('whileUpload') != null && document.getElementById('whileUpload') != undefined) {
-                document.getElementById('whileUpload').style.display = "none";
+
+    function afterMediatorUpdate(src) {
+        var error  = src.trim();
+        if(error != '' && (error.indexOf("error:") != -1)){
```
Comment (lafernando) [inline components/mediation-ui/org.wso2.carbon.sequences.ui/src/main/resources/web/sequences/design_sequence.jsp]: Check code formatting.

**Pull request merged**